### ISSUE TITLE
UTC offset of a datetime value

### ISSUE THREAD

**Issue by hhell** [ClickHouse/ClickHouse#19768]
Given a datetime value, e.g. `toDateTime('2020-01-01 00:00:00', 'America/New_York') as source_value`,
how to get the UTC offset of the said value?

One inefficient but working way to do this is as such: `toDateTime(formatDateTime(source_value, '%Y-%m-%d %H:%M:%S'), 'UTC') - source_value as utc_offset`
Is there a better way?

Related: how to reinterpret a datetime value as tz-naive? Or, failing that, as UTC datetime? The expression `toDateTime(formatDateTime(source_value, '%Y-%m-%d %H:%M:%S'), 'UTC')` does exactly that, and it is the ugly part of the expression above.

**Comment by den-crane** [comment]
yes, the only way now:

```sql
SELECT (toDateTime('2020-01-01 00:00:00', 'America/New_York') AS x) - toDateTime(toString(x), 'UTC') AS y

┌─────y─┐
│ 18000 │
└───────┘
```

**Comment by den-crane** [comment]
New feature has been implemented

```
SELECT timezoneOffset(toDateTime('2018-08-21 22:20:00', 'Australia/Lord_Howe'));
```
afaik will be available in CH 21.3+
